scapy/layers/inet6.py

```diff
@@ -1084,14 +1087,14 @@ def defragment6(pktlist):
     Crap is dropped. What lacks is completed by 'X' characters.
     """
 
-    l = filter(lambda x: IPv6ExtHdrFragment in x, pktlist) # remove non fragments
+    l = list(filter(lambda x: IPv6ExtHdrFragment in x, pktlist)) # remove non fragments
```

Comment:
Don't change the filter() calls, I will push a PR that rewritten them as list comprehensions.